support/azureutil/azureutil.go

```diff
@@ -0,0 +1,118 @@
+package azureutil
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+)
+
+// GetSubnetNameFromSubnetID extracts the subnet name from a subnet ID
+// Example subnet ID: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>/subnets/<subnetName>
+func GetSubnetNameFromSubnetID(subnetID string) (string, error) {
+	subnet, err := arm.ParseResourceID(subnetID)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse subnet ID %q: %v", subnetID, err)
+	}
+
+	if subnet.Name == "" {
+		return "", fmt.Errorf("failed to parse subnet name from %q", subnetID)
+	}
+
+	return subnet.Name, nil
+}
+
+// GetNetworkSecurityGroupNameFromNetworkSecurityGroupID extracts the network security group (nsg) name from a nsg ID
+// Example nsg ID: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<nsgName>
+func GetNetworkSecurityGroupNameFromNetworkSecurityGroupID(nsgID string) (string, error) {
```

Comment:
Should you assert type in here?  To make sure what was actually passed was a resource type of `networkSecurityGroup` ?

Reply:
Or is validation happening elsewhere?

Reply:
Something like this. 

```
res, err := arm.ParseResourceId()
if err != nil {
  return err 
} 

if !strings.EqualFold(res.ResourceType.Type, "virtualnetworks/subnets") {
  return fmt.Errorf("Invalid resource type '%s', expected 'virtualNetworks/subnets'", res.ResourceType.Type)

Reply:
Added